src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -805,5 +832,21 @@ private void ValidateSize(long availableSizeBytes, long sizeToBeReadBytes)
                     subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorProcessingData);
             }
         }
+
+        private void AddDbExecutionTimeToMiddlewareContext(long time)
+        {
+            HttpContext? httpContext = HttpContextAccessor.HttpContext;
+            if (httpContext != null)
+            {
+
+                if (!httpContext.Items.ContainsKey(TOTALDBEXECUTIONTIME))
+                {
+                    httpContext.Items[TOTALDBEXECUTIONTIME] = (long)0;
+                }
+
+                long currentTime = (long)httpContext.Items[TOTALDBEXECUTIONTIME]!;
+                httpContext.Items[TOTALDBEXECUTIONTIME] = currentTime + time;
```

Comment:
How does this work when the execution involves multiple source calls?

Reply:
Fixed.